src/actions/walletConnectActions.js

```diff
@@ -510,7 +550,18 @@ const loadLegacySessions = async (): Promise<Session[]> => {
 
 export const initWalletConnectSessions = () => {
   return async (dispatch: Dispatch, getState: GetState) => {
-    const { walletConnectSessions: { isImported, sessions } } = getState();
+    const {
+      walletConnectSessions: { isImported, sessions },
+      accounts: { data: accounts },
+    } = getState();
+
+    const keyWallet = findKeyBasedAccount(accounts);
+    if (!keyWallet) return;
+    const keyWalletAddress = getAccountAddress(keyWallet);
+
+    if (shouldClearWCSessions(sessions, keyWalletAddress)) {
```

Comment:
Shall we check if the user has a smart account?

Reply:
imho we shouldn't. If user doesn't have a SW and we leave the WC connected with key wallet, then sending transactions etc would break because we assume that user is connected via smart wallet. I've just removed the code for fetching fees for key wallet. 
If user doesn't have a smart account and we remove WC connection then nothing breaks. When user tries to connect without SW he gets an error message and we encourage him to set up SW.

Reply:
Agree, we should show then this encouraging message somewhere